pkgs/development/python-modules/qiskit/default.nix

```
@@ -1,77 +1,109 @@
 {
+  stdenv,
   lib,
   pythonOlder,
   buildPythonPackage,
   fetchFromGitHub,
+  cargo,
+  rustPlatform,
+  rustc,
+  libiconv,
 
-  # build-system
-  setuptools,
+  dill,
+  numpy,
+  python-dateutil,
+  rustworkx,
+  scipy,
+  setuptools-rust,
+  stevedore,
+  symengine,
+  sympy,
+  typing-extensions,
 
-  # Python Inputs
-  qiskit-aer,
-  qiskit-ibmq-provider,
-  qiskit-ignis,
-  qiskit-terra,
-  # Optional inputs
-  withOptionalPackages ? true,
-  qiskit-finance,
-  qiskit-machine-learning,
-  qiskit-nature,
-  qiskit-optimization,
-  # Check Inputs
-  pytestCheckHook,
+  withVisualization ? false,
+  matplotlib,
+  pillow,
+  pydot,
+  pylatexenc,
+  seaborn,
+
+  withCrosstalkPass ? false,
+  z3-solver,
+
+  withCspLayoutPass ? false,
+  python-constraint,
 }:
 
 let
-  optionalQiskitPackages = [
-    qiskit-finance
-    qiskit-machine-learning
-    qiskit-nature
-    qiskit-optimization
+  visualizationPackages = [
+    matplotlib
+    pillow
+    pydot
+    pylatexenc
+    seaborn
   ];
+  crosstalkPackages = [ z3-solver ];
+  cspLayoutPackages = [ python-constraint ];
 in
+
 buildPythonPackage rec {
   pname = "qiskit";
-  # NOTE: This version denotes a specific set of subpackages. See https://qiskit.org/documentation/release_notes.html#version-history
-  version = "2.1.1";
-  pyproject = true;
+  version = "2.1.2";
+  format = "setuptools";
 
-  disabled = pythonOlder "3.6";
+  disabled = pythonOlder "3.9";
 
   src = fetchFromGitHub {
     owner = "Qiskit";
     repo = "qiskit";
-    tag = version;
-    hash = "sha256-WHfsl/T4lmnvkGY7gF5PStilGq3G66TZG9oB1tKwuOQ=";
+    rev = "refs/tags/${version}";
```

```suggestion
    tag = version;
```
tag is doing that already